packages/lodestar/src/db/api/beacon/repositories/blockArhive.ts

```diff
@@ -0,0 +1,32 @@
+import {BeaconBlock} from "@chainsafe/eth2.0-types";
+import {IBeaconConfig} from "@chainsafe/eth2.0-config";
+import {BulkRepository} from "../repository";
+import {IDatabaseController} from "../../../controller";
+import {Bucket, encodeKey} from "../../../schema";
+import {serialize} from "@chainsafe/ssz";
+
+/**
+ * Stores finalized blocks. Block slot is identifier.
+ */
+export class BlockArchiveRepository extends BulkRepository<BeaconBlock> {
+
+  public constructor(
+    config: IBeaconConfig,
+    db: IDatabaseController
+  ) {
+    super(config, db, Bucket.blockArchive, config.types.BeaconBlock);
+  }
+
+  public async addMultiple(blocks: BeaconBlock[]): Promise<void> {
+    await this.db.batchPut(
+      blocks.map((block) => ({
+        key: encodeKey(this.bucket, block.slot),
+        value: serialize(block, this.type)
+      }))
+    );
+  }
+
+  public async setUnderRoot(value: BeaconBlock): Promise<void> {
```

Comment:
I think we should rename `setUnderRoot` (everywhere) to something more generic, like `add` (also goes well with `addMultiple`)
Maybe we can also break it out into two functions:
``
```
createId(value: T): Id;

// and

add(value: T): Promise<void> {
  this.set(this.createId(value), value);
}
```
We probably would mostly just need to override `createId`.

We should also merge the naming disparities between `deleteMany` and `addMultiple`.

Reply:
I think we should leave set under root as it's quite clear what's going on. We have method where you can set id whatever you wan't, but this is extremely common scenario where we store under hashTreeRoot or signingRoot. We can create another method with custom id function though. 

> We should also merge the naming disparities between deleteMany and addMultiple
agree

Reply:
Its actually not clear. (I was staring at this for a few minutes, and I'm sure other people will do the same) It assumes that all indices are roots.
And doesn't lend any additional clarity (eg: why should the chain care that the block was stored under a root? It just wants to add the block to the db)

I'm imagining a nice generic interface like:
```
createId(value: T): Id;

add(value: T): Promise<void>;
remove(value: T): Promise<void>;
bulkAdd(values: T[]): Promise<void>;
bulkRemove(values: T[]): Promise<void>;

set(id: Id, value: T): Promise<void>;
delete(id: Id): Promise<void>;
bulkSet(ids: Id[], values: T[]): Promise<void>;
bulkDelete(ids: Id[]): Promise<void>;
```

Reply:
@wemeetagain is right. I thought it was indexed by roots 😂 until i read it closer

Reply:
it is indexed by root 😂

Reply:
I like that interface as well.

Reply:
is it not by slot there?

Reply:
oh yeah, thought we are talking about block bucket 😂